contracts/FractalModule.sol

```diff
@@ -0,0 +1,72 @@
+pragma solidity ^0.8.0;
+
+import "@gnosis.pm/zodiac/contracts/core/Module.sol";
+import "@gnosis.pm/safe-contracts/contracts/common/Enum.sol";
+
+contract FractalModule is Module {
+    mapping(address => bool) public controllers; // A DAO may authorize users to act on the behalf of the parent DAO.
+    event ControllersAdded(address[] controllers);
+    event ControllersRemoved(address[] controllers);
+
+    /**
+     * @dev Throws if called by any account other than the owner.
+     */
+    modifier OnlyAuthorized() {
+        require(
+            owner() == msg.sender || controllers[msg.sender],
+            "Not Authorized"
+        );
+        _;
+    }
+
+    /// @dev Initialize function
+    /// @param initializeParams Parameters of initialization encoded
+    function setUp(bytes memory initializeParams) public override initializer {
+        __Ownable_init();
+        (
+            address _owner, // Controlling DAO
+            address _avatar, // GSafe // Address(0) == msg.sender
+            address _target, // GSafe or Modifier  // Address(0) == msg.sender
+            address[] memory _controllers // Authorized controllers
+        ) = abi.decode(
+                initializeParams,
+                (address, address, address, address[])
+            );
+    
+        setAvatar(_avatar == address(0) ? msg.sender : _avatar);
+        setTarget(_target == address(0) ? msg.sender : _target);
+        addControllers(_controllers);
+        transferOwnership(_owner);
+    }
+
+    function batchExecTxs(
+        address[] memory targets,
+        uint256[] memory values,
+        bytes[] memory datas,
+        Enum.Operation[] memory operations
+    ) public OnlyAuthorized {
+        for (uint256 i; i < targets.length; i++) {
+            require(
+                exec(targets[i], values[i], datas[i], operations[i]),
+                "Module transaction failed"
+            );
+        }
+    }
```

Comment:
As per phone discussion, let's make this function just take single instances of data, instead of arrays. We'll rely on a client calling through into a `MultiSend` contract to run batches.